packages/repository/src/relations/relation.helpers.ts

```diff
@@ -106,20 +109,54 @@ export async function includeRelatedModels<
     throw err;
   }
 
+  const relationKeys = {} as {[k in keyof T]: true};
+  const pruningMask = {} as {[k in keyof T]: undefined};
+
+  const [entityPromise, resolveEntitiesWithFK] = captureResult(() => {
+    const fields = Object.keys(relationKeys) as (keyof T)[];
+    const {filter: newFilter, fieldsAdded} = ensureFields(fields, filter ?? {});
+
+    fieldsAdded.forEach(f => (pruningMask[f] = undefined));
+    return resolveEntities(newFilter);
+  });
```

Comment:
I am struggling to understand what is this code doing and what is the purpose of `captureResult` and `captureInvocation`. Let's try to find a simpler (less clever/smart) solution please, one that's easier to understand even to casual readers.